File: algorithms/229.Majority_Element_II/Majority_Element_II.java
<<<<<<< HEAD
public List<Integer> majorityElement(int[] nums) {
    List<Integer> result=new ArrayList<Integer>();
    int y = 0, z = 1, cy = 0, cz = 0;
    for (int x: nums) {
      if (x==y) 
        cy++;
      else if (x==z) 
        cz++;
      else if (cy==0){
          y=x; 
          cy=1;
      }else if (cz==0){
          z=x; 
          cz=1;
      }else{ 
          cy--;
          cz--;
      }
    }
    cy = cz = 0;
    for (int x: nums){
      if (x==y) 
        cy++;
      else if (x== z) 
        cz++;
    }
    if (cy>nums.length/3) 
        result.add(y);
    if (cz>nums.length/3) 
        result.add(z);
    return result;
=======
public List<Integer> majorityElement(int[] nums) {
    List<Integer> result=new ArrayList<Integer>();
    int y = 0, z = 1, cy = 0, cz = 0;
    for (int x: nums) {
      if (x==y) 
        cy++;
      else if (x==z) 
        cz++;
      else if (cy==0){
          y=x; 
          cy=1;
      }else if (cz==0){
          z=x; 
          cz=1;
      }else{ 
          cy--;
          cz--;
      }
    }
    cy = cz = 0;
    for (int x: nums){
      if (x==y) 
        cy++;
      else if (x== z) 
        cz++;
    }
    if (cy>nums.length/3) 
        result.add(y);
    if (cz>nums.length/3) 
        result.add(z);
    return result;
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
}

File: algorithms/229.Majority_Element_II/Majority_Element_II.cpp
<<<<<<< HEAD
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  vector<int> majorityElement(vector<int> &a) {
    int y = 0, z = 1, cy = 0, cz = 0;
    for (auto x: a) {
      if (x == y) cy++;
      else if (x == z) cz++;
      else if (! cy) y = x, cy = 1;
      else if (! cz) z = x, cz = 1;
      else cy--, cz--;
    }
    cy = cz = 0;
    for (auto x: a)
      if (x == y) cy++;
      else if (x == z) cz++;
    vector<int> r;
    if (cy > a.size()/3) r.push_back(y);
    if (cz > a.size()/3) r.push_back(z);
    return r;
  }
};

int main() {

    return 0;
}

=======
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  vector<int> majorityElement(vector<int> &a) {
    int y = 0, z = 1, cy = 0, cz = 0;
    for (auto x: a) {
      if (x == y) cy++;
      else if (x == z) cz++;
      else if (! cy) y = x, cy = 1;
      else if (! cz) z = x, cz = 1;
      else cy--, cz--;
    }
    cy = cz = 0;
    for (auto x: a)
      if (x == y) cy++;
      else if (x == z) cz++;
    vector<int> r;
    if (cy > a.size()/3) r.push_back(y);
    if (cz > a.size()/3) r.push_back(z);
    return r;
  }
};

int main() {

    return 0;
}

>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
